### PR TITLE
Feature/animal module - A Miscellanea 

### PR DESCRIPTION
**Pull request by TaranRallings**
# Description

This is a small PR with a miscellaneous bag of changes that combines housekeeping with some minor updates that add bit of realism to certain methods. 

Changes include:
- communities now know what grid they are in and what grids are neighbouring 
- an initialization method to populate communities with one cohort of each kind
- natural mortality functionality based on allometric longevity
- energetically motivated dispersal
- tests

Fixes # (issue)

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Key checklist

- [x] Make sure you've run the `pre-commit` checks: `$ pre-commit run -a`
- [x] All tests pass: `$ poetry run pytest`

## Further checks

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: virtual_rainforest/models/animals/animal_cohorts.py [hunk 1]
```diff
@@ -303,3 +307,42 @@ def can_reproduce(self) -> bool:
 
         """
         return self.stored_energy >= self.reproduction_energy_threshold
+
+    def is_below_energy_threshold(self) -> bool:
+        """Check if cohort's energy is below a certain threshold.
```
Comment (jacobcook1995) [inline virtual_rainforest/models/animals/animal_cohorts.py:312]: Is this only used for migrating cohorts? if so I would be more explicit about it in the docstring

File: virtual_rainforest/models/animals/animal_cohorts.py [hunk 2]
```diff
@@ -303,3 +307,42 @@ def can_reproduce(self) -> bool:
...
+
+        Return:
+            A bool of whether the current energy state is above the migration threshold.
+        """
+        return (
+            self.stored_energy
+            < ENERGY_PERCENTILE_THRESHOLD * self.reproduction_energy_threshold
```
Comment (jacobcook1995) [inline virtual_rainforest/models/animals/animal_cohorts.py:319]: Ahh the use of `self.reproduction_energy_threshold` made me think that this was for migratory birds, is reproduction energy just used as a convenient stand in?

File: tests/models/animals/test_animal_communities.py [hunk 3]
```diff
@@ -17,19 +17,41 @@ def functional_group_list_instance(shared_datadir):
 
 
 @pytest.fixture
-def animal_community_instance(functional_group_list_instance):
+def animal_model_instance(data_instance, functional_group_list_instance):
```
Comment (jacobcook1995) [inline tests/models/animals/test_animal_communities.py:20]: If you use this fixture across multiple scripts just put a copy of it in `tests/conftest.py` and then it becomes accessible everywhere

File: tests/models/animals/test_animal_model.py [hunk 4]
```diff
@@ -23,6 +23,18 @@ def functional_group_list_instance(shared_datadir):
     return fg_list
 
 
+@pytest.fixture
```
Comment (jacobcook1995) [inline tests/models/animals/test_animal_model.py:26]: Yeah add this fixture to `contest.py` so you don't need to add it in multiple places (as it's a fixture you don't need to import it either)

File: tests/models/animals/test_dummy_plants_and_soil.py [hunk 5]
```diff
@@ -3,12 +3,41 @@
...
+        import_functional_groups,
+    )
+
+    file = shared_datadir / "example_functional_group_import.csv"
+    fg_list = import_functional_groups(file)
+
+    return fg_list[3]
+
+
+@pytest.fixture
```
Comment (jacobcook1995) [inline tests/models/animals/test_dummy_plants_and_soil.py:35]: Feel like this fixture might also belong in `contest.py`

**Review by jacobcook1995 — APPROVED** [review]
LGTM! Just had a few minor comments

File: virtual_rainforest/models/animals/animal_cohorts.py (hunk 2 above)
Comment (TaranRallings) [reply virtual_rainforest/models/animals/animal_cohorts.py:319]: It is just a stand-in. I'll update the docs to be explicit.

File: tests/models/animals/test_animal_communities.py (hunk 3 above)
Comment (davidorme) [reply tests/models/animals/test_animal_communities.py:20]: Note that you can also have `tests/models/animals/conftest.py` to share animal model specific fixtures and keep the root conftest clean.

**Pull request merged**